src/util/codec/bytes.rs

```diff
@@ -70,13 +94,17 @@ pub trait BytesDecoder: NumberDecoder {
     /// Get the remaining length in bytes of current reader.
     fn remaining(&self) -> usize;
 
-    fn decode_bytes(&mut self) -> Result<Vec<u8>> {
+    fn decode_bytes(&mut self, desc: bool) -> Result<Vec<u8>> {
```

Comment:
Looks mismatching.
And I see that for number, you have desc decode function too.